hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledTaskResult.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.scheduledexecutor.impl;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+
+import java.io.IOException;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.ExecutionException;
+
+public class ScheduledTaskResult
```

Comment:
Just an idea, you don't need to use it if you don't like it - instead of having a Future and a resultRef in the `ScheduledTaskDescriptor`, maybe on reconstruction we can just put a future which has already completed - something like `java.util.concurrent.CompletableFuture#completedFuture` or like Guava's `Futures.immediate*`.